app/models/workers_scaling/algorithms/resources_usage_minimization.rb

```diff
@@ -21,33 +21,32 @@ def self.description
     TOLERANCE = 10
 
     ##
-    # Schedules one Worker on each available infrastructure if Experiment size is greater than infrastructures number
-    # Otherwise uses random subset of available infrastructures with size equal to Experiment size
+    # Schedules one Worker on each available configuration if Experiment size is greater than configurations number
+    # Otherwise uses random subset of available configurations with size equal to Experiment size
     def initial_deployment
-      LOGGER.debug 'Initial deployment'
+      log(:debug, 'Initial deployment')
       @experiment.reload
-      @resources_interface.get_available_infrastructures
-          .select { |infrastructure| @resources_interface.current_infrastructure_limit(infrastructure) > 0 }
+      @resources_interface.get_available_resource_configurations
+          .select { |configuration| @resources_interface.current_resource_configuration_limit(configuration) > 0 }
           .shuffle[0..@experiment.size-1]
-          .each do |infrastructure|
-        @resources_interface.schedule_workers(1, infrastructure)
-        LOGGER.debug "Initializing infrastructure: #{infrastructure}"
+          .each do |configuration|
+        @resources_interface.schedule_workers(1, configuration)
+        log(:debug, "Initializing configuration: #{configuration}")
       end
     end
 
     ##
     # Executes action based on result of time_constraint_check
-    def experiment_status_check
-      LOGGER.debug 'experiment_status_check'
+    def execute_algorithm_step
       @experiment.reload
-      current_makespan = @experiment_metrics.makespan(cond: Query::Workers::RUNNING_WITH_FINISHED_SIMULATIONS)
-      case time_constraint_check(current_makespan, planned_finish_time - Time.now)
+      current_makespan = @experiment_metrics.makespan
+      case time_constraint_check(current_makespan, self.planned_finish_time - Time.now)
         when :increase
           increase_computational_power
         when :decrease
           decrease_computational_power
         else
```

Comment:
W tym przypadku nie robi to żadnej różnicy, :ok oznacza "w pozostałym przypadku".